src/ui/components/ConfirmButton/index.js

```diff
@@ -116,11 +117,18 @@ export class ConfirmButtonBase extends React.Component<InternalProps> {
   }
 }
 
+export const extractId = (ownProps: Props) => {
+  const buttonId =
+    ownProps.id ||
+    `${ownProps.className || ''}-${ownProps.message.replace(/\s/g, '')}`;
```

Comment:
It might be helpful to keep the ID readable so I'd suggest replacing spaces with underscores: `ownProps.message.replace(/\s/g, '_')`

Reply:
Honestly, I'd rather make `id` required than playing with props like this. It will be unreadable in the logs/devtools 🙁

Reply:
Okay, I'm fine with making `id` required. It will force the caller to think about uniqueness which is probably a good thing.

Reply:
okay, sure thing. I can make id required..